server/src/main/java/com/linecorp/centraldogma/server/internal/api/CredentialServiceV1.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package com.linecorp.centraldogma.server.internal.api;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import com.linecorp.armeria.server.annotation.ConsumesJson;
+import com.linecorp.armeria.server.annotation.ExceptionHandler;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.MatchesParam;
+import com.linecorp.armeria.server.annotation.Param;
+import com.linecorp.armeria.server.annotation.Post;
+import com.linecorp.armeria.server.annotation.ProducesJson;
+import com.linecorp.armeria.server.annotation.Put;
+import com.linecorp.armeria.server.annotation.StatusCode;
+import com.linecorp.centraldogma.common.Author;
+import com.linecorp.centraldogma.common.EntryNotFoundException;
+import com.linecorp.centraldogma.common.Revision;
+import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.internal.api.auth.RequiresRole;
+import com.linecorp.centraldogma.server.metadata.MetadataService;
+import com.linecorp.centraldogma.server.metadata.ProjectRole;
+import com.linecorp.centraldogma.server.mirror.MirrorCredential;
+import com.linecorp.centraldogma.server.storage.project.ProjectManager;
+
+/**
+ * Annotated service object for managing credential service.
+ */
+@ProducesJson
+@RequiresRole(roles = ProjectRole.OWNER)
+@ExceptionHandler(HttpApiExceptionHandler.class)
+public class CredentialServiceV1 extends AbstractService {
+
+    private final MetadataService mds;
+
+    public CredentialServiceV1(ProjectManager pm, CommandExecutor executor, MetadataService mds) {
+        super(pm, executor);
+        this.mds = mds;
+    }
+
+    /**
+     * GET /projects/{projectName}/credentials
+     *
+     * <p>Returns the list of the credentials in the project.
+     */
+    @Get("/projects/{projectName}/credentials")
+    public CompletableFuture<List<MirrorCredential>> listCredentials(@Param String projectName) {
+        return projectManager()
+                .get(projectName)
+                .metaRepo()
+                .credentials();
+    }
+
+    /**
+     * GET /projects/{projectName}/credentials?id={id}
+     *
+     * <p>Returns the credential for the ID in the project.
+     */
+    // TODO(ikhoon): Automatically inject the non-optional query parameters to Route.matchesParam().
+    @MatchesParam("id")
+    @Get("/projects/{projectName}/credentials")
+    public CompletableFuture<MirrorCredential> getCredentialById(@Param String projectName, @Param String id) {
```

Comment:
Question) Are `credentialId`s unique? I wasn't able to find any checks for duplicates, and I think users can still manually modify the `credentials.json` file anyways

I wasn't able to check the UI, but I'm wondering what happens if
1) a credential doesn't have an id
2) a credential has a duplicate id